docs/howtos/telemetry/40-custom-otel-collector.md

```diff
@@ -0,0 +1,334 @@
+doc-persona: [kubewarden-operator, kubewarden-integrator]
+doc-type: [howto]
+doc-topic: [operator-manual, telemetry, metrics, quick-start]
+---
+
+<head>
+  <link rel="canonical" href="https://docs.kubewarden.io/howtos/telemetry/custom-otel-collector"/>
+</head>
+
+This guide explains how Kubewarden can be configured to send telemetry data to an OpenTelemetry collector
+that has been previously deployed on the cluster.
+
+We will deploy only one instance of the [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/)
+inside of the cluster.
+
+## Install dependencies
+
+First, we begin by installing the dependencies of OpenTelemetry Collector.
+
+We want the communication between the Kubewarden components and the collector to be encrypted.
+Hence we will leverage [cert-manager](https://cert-manager.io/) to manage all the certificates
+required to secure the communication.
+
+The traces collected by the OpenTelemetry Collector will be sent to a [Jaeger](https://www.jaegertracing.io/)
+instance.
+
+The Kubewarden stack will send metrics to the OpenTelemetry Collector. This one will expose the metrics
+as a Prometheus endpoint. The metrics will then be scraped by a Prometheus instance and stored in its
+database. The same Prometheus instance will also expose a UI to interact with the metrics.
+
+Some of the resources we will create are going to be defined inside of the `kubewarden`
+Namespace, or expect its existence. Because of that, we begin by creating the Namespace:
+
+```console
+kubectl create namespace kubewarden
+```
+
+### Install cert-manager and OpenTelemetry
+
+cert-manager and OpenTelemetry operator can be installed in this way:
+
+```console
+helm repo add jetstack https://charts.jetstack.io
+helm install --wait \
+    --namespace cert-manager \
+    --create-namespace \
+    --set crds.enabled=true \
+    --version 1.15.1 \
+    cert-manager jetstack/cert-manager
+
+
+helm repo add open-telemetry https://open-telemetry.github.io/opentelemetry-helm-charts
+helm install --wait \
+  --namespace open-telemetry \
+  --create-namespace \
+  --version 0.65.0 \
+  --set "manager.collectorImage.repository=otel/opentelemetry-collector-contrib" \
+  my-opentelemetry-operator open-telemetry/opentelemetry-operator
+``` 
+
+
+The communication between the Kubewarden components and the OpenTelemetry Collector will be
+established using mTLS.
+
+To do that, we need to create the whole PKI infrastructure:
+
+```yaml
+# pki.yaml file
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: my-client-certificate
+  namespace: kubewarden
+spec:
+  dnsNames:
+  - kubewarden.kubewarden.svc
+  - kubewarden.kubewarden.svc.cluster.local
+  issuerRef:
+    kind: Issuer
+    name: my-client-selfsigned-issuer
+  secretName: my-client-cert
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: my-client-selfsigned-issuer
+  namespace: kubewarden
+spec:
+  selfSigned: {}
```

Comment:
@jvanz I was wondering if we actually need to create two different issuers. If I'm not mistaken, the issuer leads to the generation of a self-signed CA. If that's true, the same CA could be used to issue the certificates of everything

Reply:
While possible, I'm ok with having 2 CA authorities in the shape of 2 Issuers, it signifies that the actors on the Kubewarden side and on the OTLP may be separate teams.

Reply:
As far as I know, it's possible to use the same CA. I've used two different ones just to demonstrate that they are two separated entities and to ensure that the Otel configuration was working as expected during my tests ( I've tested changing the CA in the collector configuration to ensure it was right )

Reply:
Thinking about it, it may be good to change the name of the Issuers to something that showcases that easier.